pkg/gofr/datasource/pubsub/nats/errors.go

```diff
@@ -0,0 +1,24 @@
+package nats
```

Comment:
All the errors defined here are keep exported, but i think we can make the errors unexported and only export the ones needed since they will not be used by user directly

Reply:
needed for testing

Reply:
@mfreeman451 if you want to write integration tests outside of the package you should do it in examples. Please specify where exactly are you using these exported errors for testing outside the package `nats`

Reply:
The contribution guidelines state that new integrations should have test coverage, this is a go idiomatic practice to place your tests in their own package.